src/NuGetForUnity/Editor/PathHelper.cs

```diff
@@ -0,0 +1,223 @@
+﻿using System;
+using System.IO;
+using System.Runtime.CompilerServices;
+using System.Text;
+using UnityEngine;
+
+/// <summary>
+///     Helper class to handle path operations.
+/// </summary>
+internal static class PathHelper
+{
+    /// <summary>
+    ///     The path comparison type to use for the current platform the editor is running on.
+    /// </summary>
+    internal static readonly StringComparison PathComparisonType =
+        Application.platform == RuntimePlatform.WindowsEditor || Application.platform == RuntimePlatform.OSXEditor ?
+            StringComparison.OrdinalIgnoreCase :
+            StringComparison.Ordinal;
+
+    /// <summary>
+    ///     Create a relative path from one path to another. Paths will be resolved before calculating the difference.
+    ///     Default path comparison for the active platform will be used (OrdinalIgnoreCase for Windows or Mac, Ordinal for Unix).
+    /// </summary>
+    /// <remarks>
+    ///     Should the project upgrade to Unity version that has .NET Standard 2.1, this can be removed and instead we will use:
+    ///     Path.GetRelativePath(relativeTo, path).
+    ///     <see
+    ///         href="https://github.com/dotnet/runtime/blob/4aa77dfb0674fd61e41e574b8cc8ae63146f52c4/src/libraries/System.Private.CoreLib/src/System/IO/Path.cs#L843" />
+    ///     .
+    /// </remarks>
+    /// <param name="relativeTo">The source path the output should be relative to. This path is always considered to be a directory.</param>
+    /// <param name="path">The destination path.</param>
+    /// <returns>The relative path or <paramref name="path" /> if the paths don't share the same root.</returns>
+    internal static string GetRelativePath(string relativeTo, string path)
+    {
+        if (string.IsNullOrEmpty(relativeTo))
+        {
+            throw new ArgumentNullException(nameof(relativeTo));
+        }
+
+        if (string.IsNullOrEmpty(path))
+        {
+            return path;
+        }
+
+        relativeTo = Path.GetFullPath(relativeTo);
```

Comment:
We also can not use Path.GetFullPath since it will not return what we need when calling through CLI version with a path to the project passed as argument.